src/desktop/space/element.rs

```diff
@@ -10,6 +10,33 @@ use std::{
 };
 use wayland_server::protocol::wl_surface::WlSurface;
 
+/// Indicates default values for some zindexs inside smithay
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(u8)]
+pub enum RenderZindex {
+    /// WlrLayer::Background default zindex
+    Background = 10,
+    /// WlrLayer::Bottom default zindex
+    Bottom = 20,
+    /// Default zindex for Windows
+    Shell = 30,
+    /// WlrLayer::Top default zindex
+    Top = 40,
+    /// Default zindex for Windows PopUps
+    Popups = 50,
```

Comment:
So window popups are always drawn above the `wlrlayer::top layer`?

I though popups were considered as if they were part of their parent surface, as far as z-index is concerned. Why do they have their own layer?

Reply:
Well imagine a background layer having a popup (e.g. a context menu), if you can open it, your mouse cursor is obviously on the background layer. But should the menu extend below an open window? Does that make any sense?

The xdg-shell protocol does not define how they should be stacked (except in relation to other xdg-popups of the same toplevel), but I think this is the expected behavior.

Reply:
Right, good point.